code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -264,6 +264,46 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
 
 	ui_interact(user)
 
+
+//  Cafe stuff, hopefully not broken as hell
+
+/obj/machinery/chem_dispenser/brewer
+	name = "Space-Brewery"
+	icon_state = "brewer"
+	var/obj/item/weapon/reagent_containers/glass/kettle = null
+	dispensable_reagents = list("tea","greentea","redtea", "coffee","milk","cream","water","hot_coco",
+	"soymilk")
+
+
+/obj/machinery/chem_dispenser/brewer/mapping
+	max_energy = 100
+	energy = 100
+
+/obj/machinery/chem_dispenser/brewer/togglePanelOpen(var/obj/toggleitem, mob/user)
+	if(beaker)
+		user << "You can't reach the maintenance panel with a kettle in the way!"
+		return
+	return ..()
+
+/obj/machinery/chem_dispenser/brewer/attackby(var/obj/item/weapon/D as obj, var/mob/user as mob) //to be worked on

```

Comment:
You should inherit the parent of this somewhere.
Try 
`if(..())`
`    return 1`
at the start of the proc.